// @growi/core@2.0.0/dist/interfaces/index.js
import { ColorScheme as e } from "./color-scheme.js";
import { getIdForRef as i, getIdStringForRef as n, isPopulated as f, isRef as p } from "./common.js";
import { ConfigSource as a, defineConfig as m } from "./config-manager.js";
import { GrowiWikiType as s } from "./growi-app-info.js";
import { GrowiThemeSchemeType as x, isGrowiThemeMetadata as l } from "./growi-theme-metadata.js";
import { AllLang as y, Lang as P } from "./lang.js";
import { GroupType as u, PageGrant as N, PageStatus as E, isIPageInfo as F, isIPageInfoForEmpty as O, isIPageInfoForEntity as G, isIPageInfoForListing as c, isIPageInfoForOperation as A, isIPageNotFoundInfo as C } from "./page.js";
import { isNonBlankString as h, isNonEmptyString as k, toNonBlankString as R, toNonBlankStringOrUndefined as U, toNonEmptyString as w, toNonEmptyStringOrUndefined as B } from "./primitive/string.js";
import { Origin as _, allOrigin as M } from "./revision.js";
import { ACTION as j, ALL_SIGN as q, SCOPE as v } from "./scope.js";
import { AllSubscriptionStatusType as D, SubscriptionStatusType as H } from "./subscription.js";
import { USER_STATUS as K } from "./user.js";
export {
  j as ACTION,
  q as ALL_SIGN,
  y as AllLang,
  D as AllSubscriptionStatusType,
  e as ColorScheme,
  a as ConfigSource,
  u as GroupType,
  x as GrowiThemeSchemeType,
  s as GrowiWikiType,
  P as Lang,
  _ as Origin,
  N as PageGrant,
  E as PageStatus,
  v as SCOPE,
  H as SubscriptionStatusType,
  K as USER_STATUS,
  M as allOrigin,
  m as defineConfig,
  i as getIdForRef,
  n as getIdStringForRef,
  l as isGrowiThemeMetadata,
  F as isIPageInfo,
  O as isIPageInfoForEmpty,
  G as isIPageInfoForEntity,
  c as isIPageInfoForListing,
  A as isIPageInfoForOperation,
  C as isIPageNotFoundInfo,
  h as isNonBlankString,
  k as isNonEmptyString,
  f as isPopulated,
  p as isRef,
  R as toNonBlankString,
  U as toNonBlankStringOrUndefined,
  w as toNonEmptyString,
  B as toNonEmptyStringOrUndefined
};
