// @growi/core@2.0.0/dist/interfaces/scope.js
const p = {
  admin: {
    top: {},
    app: {},
    security: {},
    markdown: {},
    customize: {},
    import_data: {},
    export_data: {},
    data_transfer: {},
    external_notification: {},
    slack_integration: {},
    legacy_slack_integration: {},
    user_management: {},
    user_group_management: {},
    audit_log: {},
    plugin: {},
    ai_integration: {},
    full_text_search: {}
  }
}, u = {
  user_settings: {
    info: {},
    external_account: {},
    password: {},
    api: {
      api_token: {},
      access_token: {}
    },
    in_app_notification: {},
    other: {}
  },
  features: {
    ai_assistant: {},
    page: {},
    share_link: {},
    bookmark: {},
    attachment: {},
    page_bulk_export: {}
  }
}, l = {
  ...p,
  ...u
}, E = {
  READ: "read",
  WRITE: "write"
}, f = "*", g = Object.values(E).reduce(
  (t, n) => (t[n] = l, t),
  {}
), S = () => {
  const t = {}, n = (i, _ = [], o) => {
    for (const [s, e] of Object.entries(i)) {
      const a = s.toUpperCase(), c = [..._, s], r = c.join(":");
      e != null && (typeof e == "object" && Object.keys(e).length === 0 ? o[a] = `${r}` : typeof e == "object" && (o[a] = {
        ALL: `${r}:${f}`
      }, n(e, c, o[a])));
    }
  };
  return n(g, [], t), t;
}, k = S();
export {
  E as ACTION,
  f as ALL_SIGN,
  k as SCOPE
};
