// @growi/core@2.0.0/dist/models/devided-page-path.js
var m = Object.defineProperty;
var f = (e, t, r) => t in e ? m(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r;
var o = (e, t, r) => f(e, typeof t != "symbol" ? t + "" : t, r);
import { normalizePath as c } from "../utils/path-utils.js";
const n = /^(.+\/[^/]+)\/(\d{4}|\d{4}\/\d{2}|\d{4}\/\d{2}\/\d{2})$/;
class u {
  constructor(t, r = !1, h = !1) {
    o(this, "isRoot");
    o(this, "isFormerRoot");
    o(this, "former");
    o(this, "latter");
    if (this.isRoot = !1, this.isFormerRoot = !1, this.former = "", t == null || t === "" || t === "/") {
      this.isRoot = !0, this.latter = "/";
      return;
    }
    const i = r ? t : c(t);
    if (this.latter = i, h) {
      const a = i.match(n);
      if (a != null) {
        this.former = a[1], this.latter = a[2];
        return;
      }
    }
    let l = /^((.*)\/(?!em>))?(.+)$/;
    try {
      l = new RegExp("^((.*)(?<!<)\\/)?(.+)$");
    } catch {
    }
    const s = i.match(l);
    s != null && (this.isFormerRoot = s[1] === "/", this.former = s[2], this.latter = s[3]);
  }
}
export {
  u as DevidedPagePath
};
