// @growi/pluginkit@1.2.0/dist/v4/server/utils/template/scan.js
import h from "node:fs";
import o from "node:path";
import { isTemplateStatusValid as g } from "../../../interfaces/template.js";
import { getStatus as w } from "./get-status.js";
import { validateTemplatePluginGrowiDirective as T } from "./validate-growi-plugin-directive.js";
const v = async (a, e, u, i) => {
  const l = [], c = o.resolve(a, "dist", e);
  let d = !1;
  for await (const n of u.supportingLocales) {
    const r = o.resolve(c, n);
    try {
      const t = await w(r), { isTemplateExists: m, meta: s } = t;
      if (!m) throw new Error("'template.md does not exist.");
      if (s == null) throw new Error("'meta.md does not exist.");
      if ((s == null ? void 0 : s.title) == null)
        throw new Error("'meta.md does not contain the title.");
      const f = !d;
      l.push({
        pluginId: i == null ? void 0 : i.pluginId,
        id: e,
        locale: n,
        isValid: !0,
        isDefault: f,
        title: s.title,
        desc: s.desc
      }), d = !0;
    } catch (t) {
      l.push({
        pluginId: i == null ? void 0 : i.pluginId,
        id: e,
        locale: n,
        isValid: !1,
        invalidReason: t.message
      });
    }
  }
  return console.debug(`Template directory (${a}) has scanned`, {
    status: l
  }), l;
}, P = async (a, e) => {
  const u = (e == null ? void 0 : e.data) ?? T(a), i = [], l = o.resolve(a, "dist"), c = h.readdirSync(l);
  for await (const d of c) {
    const n = (await v(a, d, u, {
      pluginId: e == null ? void 0 : e.pluginId
    })).filter((t) => e != null && e.returnsInvalidTemplates ? !0 : t.isValid), r = n.find(
      (t) => "isDefault" in t && t.isDefault
    );
    r == null || !g(r) || i.push({
      // for the 'default' key
      default: r,
      // for each locale keys
      ...Object.fromEntries(
        n.map((t) => [t.locale, t])
      )
    });
  }
  return i;
};
export {
  P as scanAllTemplates,
  v as scanTemplate
};
