// @growi/core@2.0.0/dist/index.js
import { AcceptedUploadFileType as r } from "./consts/accepted-upload-file-type.js";
import { GrowiPluginType as i } from "./consts/growi-plugin.js";
import { GrowiDeploymentType as p, GrowiServiceType as f } from "./consts/system.js";
import { YDocStatus as g } from "./consts/ydoc-status.js";
import { ColorScheme as S } from "./interfaces/color-scheme.js";
import { getIdForRef as x, getIdStringForRef as I, isPopulated as l, isRef as y } from "./interfaces/common.js";
import { ConfigSource as d, defineConfig as u } from "./interfaces/config-manager.js";
import { GrowiWikiType as c } from "./interfaces/growi-app-info.js";
import { GrowiThemeSchemeType as N, isGrowiThemeMetadata as F } from "./interfaces/growi-theme-metadata.js";
import { AllLang as O, Lang as w } from "./interfaces/lang.js";
import { GroupType as C, PageGrant as L, PageStatus as U, isIPageInfo as h, isIPageInfoForEmpty as k, isIPageInfoForEntity as R, isIPageInfoForListing as B, isIPageInfoForOperation as b, isIPageNotFoundInfo as D } from "./interfaces/page.js";
import { isNonBlankString as v, isNonEmptyString as M, toNonBlankString as W, toNonBlankStringOrUndefined as Y, toNonEmptyString as j, toNonEmptyStringOrUndefined as q } from "./interfaces/primitive/string.js";
import { Origin as H, allOrigin as J } from "./interfaces/revision.js";
import { ACTION as Q, ALL_SIGN as V, SCOPE as X } from "./interfaces/scope.js";
import { AllSubscriptionStatusType as $, SubscriptionStatusType as oo } from "./interfaces/subscription.js";
import { USER_STATUS as ro } from "./interfaces/user.js";
export {
  Q as ACTION,
  V as ALL_SIGN,
  r as AcceptedUploadFileType,
  O as AllLang,
  $ as AllSubscriptionStatusType,
  S as ColorScheme,
  d as ConfigSource,
  C as GroupType,
  p as GrowiDeploymentType,
  i as GrowiPluginType,
  f as GrowiServiceType,
  N as GrowiThemeSchemeType,
  c as GrowiWikiType,
  w as Lang,
  H as Origin,
  L as PageGrant,
  U as PageStatus,
  X as SCOPE,
  oo as SubscriptionStatusType,
  ro as USER_STATUS,
  g as YDocStatus,
  J as allOrigin,
  u as defineConfig,
  x as getIdForRef,
  I as getIdStringForRef,
  F as isGrowiThemeMetadata,
  h as isIPageInfo,
  k as isIPageInfoForEmpty,
  R as isIPageInfoForEntity,
  B as isIPageInfoForListing,
  b as isIPageInfoForOperation,
  D as isIPageNotFoundInfo,
  v as isNonBlankString,
  M as isNonEmptyString,
  l as isPopulated,
  y as isRef,
  W as toNonBlankString,
  Y as toNonBlankStringOrUndefined,
  j as toNonEmptyString,
  q as toNonEmptyStringOrUndefined
};
