// @growi/core@2.0.0/dist/utils/path-utils.js
function e(n) {
  if (typeof n != "string" || !n || n === "") return null;
  if (n === "/")
    return {
      headingSlashes: "/",
      content: "",
      trailingSlashes: "",
      hasHeadingSlash: !0,
      hasTrailingSlash: !0
    };
  let r = 0, i = n.length;
  for (; r < n.length && n[r] === "/"; )
    r++;
  for (; i > r && n[i - 1] === "/"; )
    i--;
  const s = n.substring(0, r), u = n.substring(r, i), t = n.substring(i);
  return {
    headingSlashes: s,
    content: u,
    trailingSlashes: t,
    hasHeadingSlash: s.length > 0,
    hasTrailingSlash: t.length > 0
  };
}
function o(n) {
  if (n === "/") return !0;
  const r = e(n);
  return (r == null ? void 0 : r.hasHeadingSlash) ?? !1;
}
function g(n) {
  if (n === "/") return !0;
  const r = e(n);
  return (r == null ? void 0 : r.hasTrailingSlash) ?? !1;
}
function l(n) {
  if (n === "/") return n;
  if (n === "") return "/";
  const r = e(n);
  return r != null && r.hasHeadingSlash ? n : `/${n}`;
}
function a(n) {
  if (n === "/") return n;
  if (n === "") return "/";
  const r = e(n);
  return r != null && r.hasTrailingSlash ? n : `${n}/`;
}
function c(n) {
  if (n === "/" || n === "") return n;
  const r = e(n);
  return r != null && r.hasHeadingSlash ? n === "//" ? "/" : r.content + r.trailingSlashes : n;
}
function S(n) {
  if (n === "/" || n === "") return n;
  const r = e(n);
  return r == null ? n : r.headingSlashes + r.content;
}
function f(n) {
  if (typeof n != "string" || n === "" || n === "/")
    return "/";
  const r = e(n);
  return r == null ? "/" : `/${r.content}`;
}
function d(n) {
  return `# ${n}`;
}
function h(n, r) {
  return n === "/" ? n : l(r);
}
function H(n) {
  return f(n.split("/").slice(0, -1).join("/"));
}
export {
  l as addHeadingSlash,
  a as addTrailingSlash,
  d as attachTitleHeader,
  H as getParentPath,
  o as hasHeadingSlash,
  g as hasTrailingSlash,
  f as normalizePath,
  c as removeHeadingSlash,
  S as removeTrailingSlash,
  h as returnPathForURL
};
